### PR TITLE
fix: unify cdp approach to fix devtools in electron

### LINKED ISSUE #15932: Chromium Devtools bug when using Electron
### Current behavior

Repro:

1. launch `runner-ct` with electron
2. open devtools (right click -> inspect element)
3. run a spec
4. inspect some elements
5. run spec again
6. the devtools sometimes to not refresh

First discovered and noted here: https://github.com/cypress-io/cypress/pull/15777

### Desired behavior

Can use devtools with electron.

### Versions

This issue also occurs in E2E, running against master@6.8.0 . For me, this is occurring on every spec run.

### PR DESCRIPTION
**Pull request by ryanthemanuel**
Closes #15932

### Additional details

Electron has [issues](https://github.com/electron/electron/issues/34260) when you attach to `win.webContents.debugger` where it causes devtools to get out of sync with the DOM. This can be fixed by attaching to electron in a similar fashion to how we attach to other chromium browsers. This PR makes that transition

### Steps to test

* Ensure electron tests are passing
* Use the DOM inspector in electron and ensure that the DOM is showing up correctly

### How has the user experience changed?

n/a

### PR Tasks

- [X] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Review by chrisbreiding — APPROVED** [review]

**Review by mschile — APPROVED** [review]

File: cli/CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ _Released 04/25/2023 (PENDING)_
 
 - Fixed an issue where setting `videoCompression` to `0` would cause the video output to be broken. `0` is now treated as false. Addresses [#5191](https://github.com/cypress-io/cypress/issues/5191) and [#24595](https://github.com/cypress-io/cypress/issues/24595).
 - Fixed an issue on the [Debug page](https://on.cypress.io/debug-page) where the passing run status would appear even if the Cypress Cloud organization was over its monthly test result limit. Addresses [#26528](https://github.com/cypress-io/cypress/issues/26528).
+- Fixed an issue in Electron where devtools gets out of sync with the DOM occasionally. Addresses [#15932](https://github.com/cypress-io/cypress/issues/15932).
```
Comment (emilyrohrbough) [inline cli/CHANGELOG.md]: lets move this up. this is more important than videoCompression

Comment (ryanthemanuel) [reply cli/CHANGELOG.md]: I'll refactor this after the release today.

**Comment by lmiller1990** [comment]
I tested this out, it does indeed fix https://github.com/cypress-io/cypress/issues/15932

We could consider making the default browser electron - since it's the only one guaranteed to be on all systems. I believe this was the intention and how older Cypress versions worked, but we changed it due to this bug.

Either way, very excited to see this this is finally fixed!

File: cli/CHANGELOG.md (hunk 1 above)
Comment (ryanthemanuel) [reply cli/CHANGELOG.md]: This has been refactored

File: cli/CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,12 @@
 <!-- See the ../guides/writing-the-cypress-changelog.md for details on writing the changelog. -->
+## 12.11.1
+
+_Released 05/09/2023 (Pending)_
```
Comment (ryanthemanuel) [inline cli/CHANGELOG.md]:
```suggestion
_Released 05/09/2023 (PENDING)_
```

**Pull request merged**